assets/js/components/Dashboard/PaMessageForm/MainForm.tsx

```diff
@@ -172,6 +174,7 @@ const MainForm = ({
             setEndWithEffectPeriod={setEndWithEffectPeriod}
             selectedTemplate={selectedTemplate}
             onClearSelectedTemplate={onClearSelectedTemplate}
+            isReadOnly={isReadOnly}
```

Comment:
This is totally fine as implemented, but as a future note, using Context for global state that's used widely can save a bunch of boilerplate. In this case, if this value was passed via Context and accessed with a custom hook (e.g. `useIsPaMessageAdmin`), it could eliminate a bunch of intermediate wiring.

Reply:
Ah yeah, I did consider that approach as well! But I wasn't certain if using context was overkill somehow. I'd be happy to modify the approach to use context rather than passing it down as a prop!

Reply:
I'd say only if you think it's worth it. There's nothing wrong with explicitly passing values, and it's simpler and more explicit that way, so there are tradeoffs to consider. I just wanted to call out the alternative.

Reply:
Gotcha. I did ultimately go with this approach initially because I thought that it was more simplistic, so I'll keep it this way!